bwt/bwt.go

```diff
@@ -0,0 +1,476 @@
+package bwt
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strings"
+
+	"golang.org/x/exp/slices"
+)
+
+/*
+
+For the BWT usage, please read the BWT methods
+below. To understand what it is and how
+it works for either curiosity or maintenance, then read below.
+
+# BWT
+
+BWT Stands for (B)urrows-(W)heeler (T)ransform. The BWT aids in
+text compression and acts as a search index for any arbitrary
+sequence of characters. With the BWT and some auxiliary data
+structures, we can analyze a sequence in a memory and run time
+efficient manner.
+
+## BWT Transform
+
+The first step to build the BWT is to get the BWT itself.
+
+This is done by:
+1. Appending a null terminating character to the end of a sequence
+2. Rotate the sequence so that the last character is now the first
+3. Repeat 2. N times where N is the length of the sequence
+4. Lexicographically sort the NxN matrix of rotated sequences where
+   the null termination character is always the least-valued
+5. Store the first and last column of the matrix. The last column
+   is the output of the BWT. The first column is needed to run queries
+   on the BWT of the original sequence.
+
+Lets use banana as an example.
+
+banana$      $banana
```

Comment:
can there be an example test that takes `banana$` and returns the last column as a string?

Reply:
@TimothyStiles updated